File: decrypt_test.py
from cryptography.hazmat.primitives.kdf.pbkdf2 import PBKDF2HMAC
from cryptography.hazmat.primitives import hashes
from cryptography.hazmat.backends import default_backend
from cryptography.hazmat.primitives.ciphers.aead import AESGCM
import binascii

data = "615f01010e4c76a5-70af8049e10dca8de2106c09-f87502f6151f7f6407530ef40dcd11d86b545357f8357a160f34acb8b183907087525acfbb8c76a2"
parts = data.split("-")
salt = binascii.unhexlify(parts[0])
iv = binascii.unhexlify(parts[1])
ciphertext = binascii.unhexlify(parts[2])

password = b"player"

kdf = PBKDF2HMAC(
    algorithm=hashes.SHA256(),
    length=32,
    salt=salt,
    iterations=1000,
    backend=default_backend()
)
key = kdf.derive(password)

aesgcm = AESGCM(key)
try:
    plaintext = aesgcm.decrypt(iv, ciphertext, None)
    print("Decrypted:", plaintext.decode('utf-8'))
except Exception as e:
    print("Decryption failed:", e)

File: decrypt_test.js
const crypto = require('crypto');

const data = "615f01010e4c76a5-70af8049e10dca8de2106c09-f87502f6151f7f6407530ef40dcd11d86b545357f8357a160f34acb8b183907087525acfbb8c76a2";
const parts = data.split("-");
const salt = Buffer.from(parts[0], 'hex');
const iv = Buffer.from(parts[1], 'hex');
const ciphertext = Buffer.from(parts[2], 'hex');
const password = "player";

crypto.pbkdf2(password, salt, 1000, 32, 'sha256', (err, key) => {
    if (err) throw err;
    try {
        const decipher = crypto.createDecipheriv('aes-256-gcm', key, iv);
        // AES-GCM requires the authentication tag. In WebCrypto, the last 16 bytes of the ciphertext are the auth tag!
        const authTag = ciphertext.slice(ciphertext.length - 16);
        const encrypted = ciphertext.slice(0, ciphertext.length - 16);
        decipher.setAuthTag(authTag);
        let decrypted = decipher.update(encrypted, null, 'utf8');
        decrypted += decipher.final('utf8');
        console.log("Decrypted:", decrypted);
    } catch(e) {
        console.error("Decryption failed:", e.message);
    }
});
